components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/template/service/BallerinaOperation.java

```diff
@@ -271,11 +274,10 @@ public void setApiResponseInterceptor(String responseInterceptor) {
         }
     }
 
-    public void setBasicAuth(BasicAuth basicAuth) {
-        //update the ResourceBasicAuth property only if there is no security scheme provided during instantiation
-        if (this.basicAuth == null) {
-            this.basicAuth = basicAuth;
-            authProviders = OpenAPICodegenUtils.setAuthProviders(basicAuth);
+    public void setSecuritySchemas(String schemas) {
+        //update the Resource auth providers property only if there is no security scheme provided during instantiation
+        if (this.authProviders.size() < 1) {
```

Comment:
can we use isEmpty() here?

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/973